stdlib/InteractiveUtils/src/editless.jl

```diff
@@ -82,15 +84,16 @@ already work:
 The following defines the usage of terminal-based `emacs`:
 
     define_editor(
-        r"\\bemacs\\b.*\\s(-nw|--no-window-system)\\b", wait=true) do cmd, path, line
-        `\$cmd +\$line \$path`
+        r"\\bemacs\\b.*\\s(-nw|--no-window-system)\\b", wait=true
+    ) do cmd, path, line, wait
+        run(pipeline(`\$cmd +\$line \$path`); wait)
     end
 
 !!! compat "Julia 1.4"
     `define_editor` was introduced in Julia 1.4.
 """
 function define_editor(fn::Function, pattern; wait::Bool=false)
-    callback = function (cmd::Cmd, path::AbstractString, line::Integer, column::Integer)
+    callback = function (cmd::Cmd, path::AbstractString, line::Integer, column::Integer; wait::Bool=wait)
```

Comment:
What is the rationale for allowing different editor definitions to have different default values for `wait`? In my mind that sounds very confusing to the point where one would end up always specifying `wait` as one can't predict what value it'll have when not specified?

Reply:
Not having this also would simplify this patch, making me wonder doubly why you have it

Reply:
Ohhh, I am sorry, I see now I got it backwards: `define_editor` already has a `wait` argument. So that exists, and has the issue I mentioned (and hopefully an explanation / rationale for it somewhere). So you are making the situation better by at allowing users to control this explicitly, at least for editors defined in a way allowing for it.

Reply:
Yeah, exactly. This PR just adds the `wait` kwarg to the callback + defaults the `wait` kwarg's value to the value passed in `define_editor`. I've updated the PR description to make this more clear.